code/datums/components/anti_magic.dm

```diff
@@ -10,6 +10,10 @@
 
 /datum/component/anti_magic/Initialize(_magic = FALSE, _holy = FALSE, _psychic = FALSE, _allowed_slots, _charges, _blocks_self = TRUE, datum/callback/_reaction, datum/callback/_expire)
 	if(isitem(parent))
+		if(isorgan(parent))
+			RegisterSignal(parent, COMSIG_ORGAN_INSERTED, .proc/apply_anti_magic)
+			RegisterSignal(parent, COMSIG_ORGAN_REMOVED, .proc/on_drop)
+			allowed_slots = null //by default, organs don't work in any slots.
```

Comment:
Pass this in initialize and/or dont register for the equipping signals when the parent is an organ